common/src/com/thoughtworks/go/domain/Pipeline.java

```diff
@@ -59,7 +59,7 @@ public Pipeline(String pipelineName, BuildCause buildCause, Stage... stages) {
     }
 
     public File defaultWorkingFolder() {
-        return new File(CruiseConfig.WORKING_BASE_DIR + pipelineName);
+        return new File("" + pipelineName);
```

Comment:
There's only [one usage](https://github.com/gocd/gocd/blob/4509fe7c72bf919c33bf212752e88a45dc09bbbd/common/src/com/thoughtworks/go/domain/StageIdentifier.java#L32) of [this default constructor](https://github.com/gocd/gocd/blob/4509fe7c72bf919c33bf212752e88a45dc09bbbd/common/src/com/thoughtworks/go/domain/NullPipeline.java#L22-L24). What do you think of removing that constructor and replacing the call with

```
new NullPipeline("")
```

Then, I can remove the `"" + pipelineName` bit. If not, I'll do what you suggested. Though explicit, it makes it feel a little wrong.